src/backend/metal/src/device.rs

```diff
@@ -113,6 +113,90 @@ fn get_final_function(library: &metal::LibraryRef, entry: &str, specialization:
     Ok(mtl_function)
 }
 
+fn create_depth_stencil_state(
+    device: &metal::DeviceRef,
+    desc: &pso::DepthStencilDesc,
+) -> native::DepthStencilState {
+    let raw = metal::DepthStencilDescriptor::new();
+
+    let mut dss = native::DepthStencilState {
+        depth_stencil_desc: Some(*desc),
+        ..Default::default()
+    };
+
+    let mut all_masks_static = true;
+
+    match desc.depth {
+        pso::DepthTest::On { fun, write } => {
+            raw.set_depth_compare_function(conv::map_compare_function(fun));
+            raw.set_depth_write_enabled(write);
+        }
+        pso::DepthTest::Off => {}
+    }
+    match desc.stencil {
+        pso::StencilTest::On { ref front, ref back } => {
+            dss.stencil_front_reference = front.reference;
+            dss.stencil_back_reference = back.reference;
+
+            let front_desc = metal::StencilDescriptor::new();
+            front_desc.set_stencil_compare_function(conv::map_compare_function(front.fun));
+
+            dss.stencil_front_read_mask = front.mask_read;
+            if let pso::State::Static(mr) = front.mask_read {
```

Comment:
nit: let's just `match` instead of `if let + else`. I know this is opinionated, and I'm open to discuss it. We haven't come to a strong consensus in WebRender project, but maybe we could achieve it here? :)

Reply:
This actually started as an `if` originally, then I changed to `match`, then back to `if`, so I'm undecided too 😄